### PR TITLE
[fix](nestedtype)fix nested type with is_exclusive

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
is_exclusive in column_array/map/struct has wrong semantic , we should make sure it's nested column is also is_exclusive which can make behavior right in some operator  like join.
Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/columns/column_array.h [hunk 1]
```diff
@@ -122,6 +122,11 @@ class ColumnArray final : public COWHelper<IColumn, ColumnArray> {
     const char* get_family_name() const override { return "Array"; }
```
Comment (eldenmoon) [inline be/src/vec/columns/column_array.h:122]: what about ColumnObject?

**Review by Gabriel39 — DISMISSED** [review]

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/columns/column_object.cpp [hunk 2]
```diff
@@ -1877,6 +1877,10 @@ void ColumnObject::for_each_imutable_subcolumn(ImutableColumnCallback callback)
     }
 }
 
+bool ColumnObject::is_exclusive() const {
```
Comment (eldenmoon) [inline be/src/vec/columns/column_object.cpp:1880]: if subcolumns size is 2, then the first is returned as result, is this correct?

File: be/src/vec/columns/column_object.cpp [hunk 3]
```diff
@@ -1877,6 +1877,10 @@ void ColumnObject::for_each_imutable_subcolumn(ImutableColumnCallback callback)
...
+    for_each_imutable_subcolumn([](const auto& subcolumn) { return subcolumn.is_exclusive(); });
```
Comment (eldenmoon) [inline be/src/vec/columns/column_object.cpp:1881]: bool is_exclusive = true
for_each_imutable_subcolumn([](const auto& subcolumn) { if (!subcolumn.is_exclusive()) {is_exclusive = false} });
return is_exclusive && IColumn::is_exclusive();;

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**